notebooks/mssql2bq/mssql-to-bigquery-notebook.ipynb

```diff
@@ -0,0 +1,853 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip3 install pymssql SQLAlchemy\n",
+    "!pip3 install --upgrade google-cloud-pipeline-components kfp --user -q\n",
+    "!pip3 install google.cloud.aiplatform"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d5a7126d-fc9c-4f04-8031-cbd835f891c6",
+   "metadata": {},
+   "source": [
+    "### Once you've installed the additional packages, you may need to restart the notebook kernel so it can find the packages.\n",
+    "\n",
+    "Uncomment & Run this cell if you have installed anything from above commands"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "74245439-10f3-4631-b8cd-4c077318b737",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# import os\n",
+    "# import IPython\n",
+    "# if not os.getenv(\"IS_TESTING\"):\n",
+    "#     app = IPython.Application.instance()\n",
+    "#     app.kernel.do_shutdown(True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "50fe84b1-eb32-4cc2-a4e2-f0bd98da5b90",
+   "metadata": {},
+   "source": [
+    "# Step 2: Import Libraries"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9f225135-cfe5-4ddd-9659-ead437aa414c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import google.cloud.aiplatform as aiplatform\n",
+    "import sys\n",
+    "from kfp import dsl\n",
+    "from kfp.v2 import compiler\n",
+    "from datetime import datetime\n",
+    "import time\n",
+    "import copy\n",
+    "import json\n",
+    "import pandas as pd\n",
+    "from google_cloud_pipeline_components.experimental.dataproc import DataprocPySparkBatchOp\n",
+    "import ipywidgets as widgets\n",
+    "import sqlalchemy\n",
+    "import pymssql\n",
+    "import math"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "af0cf264-6eef-441d-a0a4-0c7bdc3d5b96",
+   "metadata": {},
+   "source": [
+    "# Step 3: Assign Parameters\n",
+    "\n",
+    "## Step 3.1 Common Parameters\n",
+    "\n",
+    "PROJECT : GCP project-id\n",
+    "\n",
+    "REGION : GCP region\n",
+    "\n",
+    "GCS_STAGING_LOCATION : GCS staging location to be used for this notebook to store artifacts\n",
+    "\n",
+    "SUBNET : VPC subnet\n",
+    "\n",
+    "JARS : list of jars. For this notebook mssql connectora and postgres connectorjar is required in addition with the dataproc template\n",
+    "\n",
+    "MAX_PARALLELISM : Parameter for number of jobs to run in parallel default value is 5"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ee427d7c-edcb-46b2-a825-624db784aff0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "PROJECT = \"<project-id>\"\n",
+    "REGION = \"<region>\"\n",
+    "GCS_STAGING_LOCATION = \"gs://<staging-bucket>\"\n",
+    "SUBNET = \"projects/<project-id>/regions/<region>/subnetworks/<subnet-name>\"\n",
+    "MAX_PARALLELISM = 5 # default value is set to 5\n",
+    "SERVICE_ACCOUNT = \"\"\n",
```

Comment:
ADD SA as SERVICE_ACCOUNT = "<service-account>"

Reply:
SA is already added as SERVICE_ACCOUNT = "" in line 168